internal/cluster/cluster.go

```diff
@@ -427,6 +427,37 @@ func (m *Manager) tryAssignMachineCidrNonDHCPMode(cluster *common.Cluster) error
 	return UpdateMachineCidr(m.db, cluster, machineCidr)
 }
 
+func (m *Manager) tryAssignMachineCidrSNO(cluster *common.Cluster) error {
+	if network.IsMachineCidrAvailable(cluster) {
+		return nil
+	}
+	clusterFamilies, err := network.CidrsToAddressFamilies(network.GetClusterNetworkCidrs(cluster))
+	if err != nil {
+		return err
+	}
+	clusterFamilies = network.CanonizeAddressFamilies(clusterFamilies)
+	serviceFamilies, err := network.CidrsToAddressFamilies(network.GetServiceNetworkCidrs(cluster))
+	if err != nil {
+		return err
+	}
+	if reflect.DeepEqual(clusterFamilies, serviceFamilies) {
```

Comment:
do the lists need to be sorted before deep equal or is it handled inside this function?

Reply:
https://github.com/ori-amizur/assisted-service/blob/c13b02e33f37ad9f8dfbc552e94ab32951410114/internal/network/utils.go#L185